Dynatrace/dynatrace-operator
# Improve oneAgent image validation using manifest labels

## Description

With this validation of the `oneAgent` image is improved. Previously the only verification that was done was by determining if there was a downgrade for the `oneAgent` or not (downgrades are not allowed). Now a new label is taken into consideration (`com.dynatrace.type`). With this we can check whether an image is immutable or not.

If the agent is immutable:
- `classicFullStack` is not allowed
- `feature.dynatrace.com/oneagent-readonly-host-fs` is not allowed

If the `com.dynatrace.type` label is missing:
- for now no error is thrown

If the `com.dynatrace.build-version` label is missing:
- throw an error

## How can this be tested?

~Deploy the operator and apply a `dynakube` (`cloudNative` or `classicFullstack`):~
~- when using the standard image everything should work normally as it a mutable image.~
~- when using the immutable image either use the feature flag (`feature.dynatrace.com/oneagent-readonly-host-fs`) or use `classicFullStack` mode: the `oneAgents` should be in an error state.~

~NOTE: For now this is not testable as there are problems with public-registry. Will be updated as soon as there is a solution.~

It should be testable in the internal e2e pipelines 😉 

## Checklist
